### ISSUE TITLE
CSS Style

### ISSUE THREAD

**Issue by madeInLagny** [torrobinson/progress-steps-webcomponent#3]
Thanks for this nice and useful webcomponent. Styling in javascript is a little restrictive. Is there a way to style the component with CSS ? Ideally using [CSS variables and customs properties ?](https://lit.dev/docs/components/styles/#theming)

**Comment by torrobinson** [comment]
Hey @madeInLagny, I agree it's restrictive - this was my first webcomponent and couldn't find any other nice way to pass styling in. I've read briefly since that it can be possible to pass css variables directly in though (that is, they can be overridden at the root externally), so let me take a look over the next day or two to see if this is possible.

I'm thinking the same css variables used internally (starting [here](https://github.com/torrobinson/progress-steps-webcomponent/blob/master/src/progress-steps.js#L49)) could be used externally if I moved where they were defined 🤔

**Comment by madeInLagny** [comment]
If this is your first webcomponent, then hats off! It's great. I would recommend using [lit](https://lit.dev/) which comes with a lot of built in features, including external styling with css variables. Good luck and thanks for the great cotribution

**Comment by torrobinson** [comment]
@madeInLagny haha thanks! I needed this component for a personal project and figured I'd learn how to make a native webcomponent just to learn it, so I haven't used any helper tools (I guess technically aside from Angular, which generates native webcomponents during a build)

It looks like css variables indeed can bleed into a shadow DOM, which is fantastic! The only thing preventing it from working with my component is that the "defaults" are defined internally, and CSS variables always pick the closest definition, so they're overridden.

What I'll do is extract the defaults into a new css file in the traditional js+css pairing and this would allow the variables to be overridden like so:
![image](https://user-images.githubusercontent.com/4401999/200026469-aa711ca4-97c1-4b31-9c13-3a7a1915df36.png)


Does that sound reasonable / do you see anything wrong with this?

If not, please trial the js/css in this new branch @ https://github.com/torrobinson/progress-steps-webcomponent/tree/feature/3-css-variable-refactor/dist
to see if this fits your needs - should be totally controllable via CSS now using the full breadth of previously internal variables:

```css
#my-steps{
	/* The color to fill up, left-to-right, as steps are set to active */
	--progress-fill-color: #cf78d9;
	/* The default color of the unfilled section of line and steps after the active step */
	--progress-unfilled-color: purple;

	/* The width of each step icon */
	--step-width: 20px;
	/* The font size of the step number and label */
	--font-size: 12px;
	/* The border radius of the step icon */
	--step-border-radius: 25%;
	/* The thickness of the line/progress bar/borders */
	--line-thickness: 3px;

	/* The animation speed of the progress bar filling up */
	--animation-speed: 500ms;
	/* Display attribute of the step labels. Show: 'inline-block', hide: 'none' */
	--step-label-display: 'inline-block';
	/* The vertical margin of the labels, if shown */
	--step-label-spacing: 5px;
	/* The font weight to use on step labels */
	--step-label-font-weight: normal;

	/* The font color of the step icon that is currently active */
	--current-step-font-color: red;
	/* The font color of the current step label */
	--current-label-font-color: blue;
	/* The font weight of the current step label */
	--current-label-font-weight: bold;

	/* The font color of step labels that are before the current step */
	--previous-label-font-color: red;
	/* The font color of the step icons that are before the current step */
	--previous-step-font-color: pink;

	/* The fill-color for step icons that are after the current active step */
	--future-step-fill-color: orange;
	/* The font color of step labels that are after the current step */
	--future-label-font-color: green;

	/* The font color of step labels that are disabled */
	--disabled-label-font-color: maroon;
	/* The font color of the disabled steps */
	--disabled-step-font-color: red;
	/* The fill-color for disabled step icons */
	--disabled-step-fill-color: blue;
}
```

**Comment by torrobinson** [comment]
Fixed with #4

**Comment by madeInLagny** [comment]
Great and thanks for your prompt and efficient support! Cheers
